components/style/properties.mako.rs

```diff
@@ -1705,35 +1705,55 @@ pub mod longhands {
         use values::computed::ComputedValueAsSpecified;
         pub use self::computed_value::T as SpecifiedValue;
 
+        const SERIF: &'static str = "serif";
+        const SANS_SERIF: &'static str = "sans-serif";
+        const CURSIVE: &'static str = "cursive";
+        const FANTASY: &'static str = "fantasy";
+        const MONOSPACE: &'static str = "monospace";
+
         impl ComputedValueAsSpecified for SpecifiedValue {}
         pub mod computed_value {
             use cssparser::ToCss;
             use std::fmt;
             use string_cache::Atom;
 
-            #[derive(Debug, PartialEq, Eq, Clone, Hash, HeapSizeOf)]
+            #[derive(Debug, PartialEq, Eq, Clone, Hash, HeapSizeOf, Deserialize, Serialize)]
             pub enum FontFamily {
                 FamilyName(Atom),
-                // Generic
-//                Serif,
-//                SansSerif,
-//                Cursive,
-//                Fantasy,
-//                Monospace,
+                // Generic,
+                Serif,
+                SansSerif,
+                Cursive,
+                Fantasy,
+                Monospace,
             }
             impl FontFamily {
                 #[inline]
                 pub fn name(&self) -> &str {
                     match *self {
                         FontFamily::FamilyName(ref name) => &*name,
+                        FontFamily::Serif => super::SERIF,
+                        FontFamily::SansSerif => super::SANS_SERIF,
+                        FontFamily::Cursive => super::CURSIVE,
+                        FontFamily::Fantasy => super::FANTASY,
+                        FontFamily::Monospace => super::MONOSPACE
+                    }
+                }
+
+                pub fn from_atom(input: Atom) -> FontFamily {
+                    match_ignore_ascii_case! { &input,
+                        super::SERIF => FontFamily::Serif,
+                        super::SANS_SERIF => FontFamily::SansSerif,
+                        super::CURSIVE => FontFamily::Cursive,
+                        super::FANTASY => FontFamily::Fantasy,
+                        super::MONOSPACE => FontFamily::Monospace
+                        _ => FontFamily::FamilyName(input.clone())
                     }
```

Comment:
Should I do this instead in order to avoid a clone?

```rust
                pub fn from_atom(input: Atom) -> FontFamily {
                    let option = match_ignore_ascii_case! { &input,
                        super::SERIF => Some(FontFamily::Serif),
                        super::SANS_SERIF => Some(FontFamily::SansSerif),
                        super::CURSIVE => Some(FontFamily::Cursive),
                        super::FANTASY => Some(FontFamily::Fantasy),
                        super::MONOSPACE => Some(FontFamily::Monospace)
                        _ => None
                    };

                    match option {
                        Some(family) => family,
                        None => FontFamily::FamilyName(input)
                    }
                }
```